shared/src/dto/mockExamAttempt/mockExamAttempt.dto.ts

```diff
@@ -0,0 +1,76 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import {
+  IsBoolean,
+  IsNotEmpty,
+  IsNumber,
+  ValidateNested,
+} from 'class-validator';
+import { CreateMockExamQuestionAttemptRequestDto } from '../mockExamQuestionAttempt/mockExamQuestionAttempt.dto';
+import { TraineeDto } from '../trainee/trainee.dto';
+
+export class DtoMockExamAttempt {
+  @ApiProperty()
+  @IsNotEmpty()
+  @IsNumber()
+  id: number;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  @ValidateNested()
+  @Type(() => TraineeDto)
+  trainee: TraineeDto;
+
+  @ApiProperty({
+    type: 'string',
+  })
+  attemptDate: ApiDate;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  @IsNumber({}, { each: true })
+  questionsIds: number[];
+
+  @ApiProperty()
+  @IsNumber()
+  score: number;
+
+  @ApiProperty()
+  @IsBoolean()
+  isPassed: boolean;
+}
+
+export class SubmitMockExamAttemptDto {
+  @ApiProperty()
+  @IsNumber()
+  traineeId: number;
+
+  @ApiProperty({ isArray: true })
```

Comment:
Let's let the swagger know what type the "question" is. Right now it suggests `null` instead of a proper object (which isn't a valid call either way)